GUI.py

```diff
@@ -200,18 +200,28 @@ def playAudio(self):
         if not self.audio.playing:
             #if self.playback_thread is None or not self.playback_thread.is_alive():
             if self.playback_thread is not None and self.playback_thread.is_alive():
-                self.playback_thread.exit()
+                #self.playback_thread.exit()
                 self.audio.stopPlayback() # Ensure stopping previous playback
+                self.playback_thread.join()
```

Comment:
Not sure if buttons respond quicker, as I am unable to test the pause/play button

Reply:
@kpothuganti please don't leave commented out code in the final version